src/honeybot/plugins/downloaded/duckduckgo/main.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+"""
+[duckduckgo.py]
+Duckduckgo Plugin
+
+[Author]
+Rakesh Seal
+
+[About]
+Returns a abstract of the search query from a duckduckgo Search. Return empty if no abstract found.
+[Website]
+
+
+[Commands]
+>>> .duck <query>
+returns abstrack from duckduckgo search
+"""
+
+import requests
+
+
+class Plugin:
+    def __init__(self):
+        pass
+    
+    def __gen_url_from_query(self, query):
+        return "https://duckduckgo.com/?q=" + query + "&format=json&pretty=1"
+
+    def run(self, incoming, methods, info, bot_info):
+        # if '!~' in info['prefix']:
+        # print(info)
+        try:
+            msgs = info["args"][1:][0].split()
+            if info['command'] == 'PRIVMSG' and msgs[0] == '.duck':
+                    query = msgs[1]
+                    duck_abstract = requests.get(Plugin.__gen_url_from_query(query)).json()["Abstract"] 
```

Comment:
Please change `Plugin.__gen...` to `self.__gen...`
Using `Plugin` directly instead of `self` does not pass the first argument `self`

You could technically use `Plugin.__gen_url_from_query(self, query)` but it is not the standard way of doing things